circup/__init__.py

```diff
@@ -1211,11 +1650,21 @@ def tags_data_save_tag(key, tag):
     message="%(prog)s, A CircuitPython module updater. Version %(version)s",
 )
 @click.pass_context
-def main(ctx, verbose, path, board_id, cpy_version):  # pragma: no cover
+def main(ctx, verbose, path, host, password, board_id, cpy_version):  # pragma: no cover
+    # pylint: disable=too-many-arguments
     """
     A tool to manage and update libraries on a CircuitPython device.
     """
     ctx.ensure_object(dict)
+
+    using_webworkflow = "host" in ctx.params.keys() and ctx.params["host"] is not None
+
+    ctx.obj["using_webworkflow"] = using_webworkflow
+    if using_webworkflow:
+        ctx.obj["backend"] = WebBackend(host=host, password=password)
+    else:
+        ctx.obj["backend"] = USBBackend()
```

Comment:
I'd expect this to take in the base path on the filesystem and the optional version string for the folder on disk case.

Reply:
latest commit changes this to accept the device path. 

Still needs a change for the version string. I'm not certain if I understand that part, but it may become more clear when I get into fixing the tests which can use these local paths I think.

Reply:
@tannewt I added `version_info` argument to the DiskBackend() constructor. The way it's coded now it's expected to be a tuple in the same form that gets returned by `get_circuitpython_version()` for example  `('8.2.8', 'adafruit_funhouse')` let me know if you prefer it be a string and if so which format it'd be expected to be in, the full contents of boot_out.txt? or something more slimmed down already?

Typically this tuple is generated by reading the contents of boot_out.txt. Perhaps I am misunderstanding it's usage, but I'm not sure that we need to provide a way to pass a mock version like this for the local folder instance. If a properly formatted boot_out.txt were put into that local folder then circup would be able to read it and get the values just as if it were a real device and then it will not rely on an argument that isn't used during normal operation. 

This PR added a mock device folder in the tests repo with a "fake" boot out text that is used for that purpose by one of the tests (which I do still need to make another pass through to fix after latest refactors) https://github.com/vladak/circup/blob/web_workflow/tests/mock_device/boot_out.txt

Reply:
I think I'd have it passed as the string argument so that parsing it is the responsibility of the disk backend.

The other way I can think to do it is the opposite, leave it backend agnostic and allow it to override whatever the backend returns.

What do you think?

Reply:
String makes sense to me.

The latest commit changes it to expect a string with the same contents formatted as `boot_out.txt` and moved the parsing functionality to a static function in Backend. Theoretically the WebBackend could make use of that as well by fetching the contents of boot_out.txt, but it makes use of the version.json API which is available in WebWorkflow instead.